src/irmin/layer.ml

```diff
@@ -0,0 +1,416 @@
+(*
```

Comment:
It might make sense to add a new package (`irmin-layer`) for this new functionality.